src/main/java/io/jenkins/plugins/checks/github/status/GitSCMSourceStatusChecksTrait.java

```diff
@@ -0,0 +1,55 @@
+package io.jenkins.plugins.checks.github.status;
+
+import hudson.Extension;
+import io.jenkins.plugins.checks.status.StatusChecksProperties;
+import jenkins.plugins.git.traits.GitSCMExtensionTrait;
+import jenkins.plugins.git.traits.GitSCMExtensionTraitDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * A trait that Warps the {@link GitSCMStatusChecksExtension} to provide {@link StatusChecksProperties}
+ * for {@link jenkins.plugins.git.GitSCMSource} job.
+ */
+@SuppressWarnings("PMD.DataClass")
+public class GitSCMSourceStatusChecksTrait extends GitSCMExtensionTrait<GitSCMStatusChecksExtension>
```

Comment:
Until I implemented this trait did I realize that we haven't supported the Git SCM Source yet, do we want that? Is this Git SCM Source popular among users? @uhafner @timja

Reply:
I thought we have only 2 different ways to use Git. Which plugin provides the Git SCM Source extension?

Reply:
I also just found out that today... 

It's also provided by Git plugin: https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/jenkins/plugins/git/GitSCMSource.java